src/Spice86.Core/Emulator/CPU/CPU.cs

```diff
@@ -40,6 +40,20 @@ public class Cpu {
     private readonly Instructions16 _instructions16;
     private readonly Instructions32 _instructions32;
     private Instructions16Or32 _instructions16Or32;
+    private int _addressSize;
+
+    /// <summary>
+    /// Address size for the currently executing instruction.
+    /// </summary>
+    public int AddressSize {
+        get => _addressSize;
+        private set {
+            if (value != 16 && value != 32)
```

Comment:
How about using an enum to avoid the if penalty at each instruction?

Reply:
Hm, I missed that it is setting it to 16 for every instruction.
That whole check can go as far as I'm concerned.